examples/cmd/testing/recordingscript.go

```diff
@@ -0,0 +1,127 @@
+package main
+
+import (
+	"bytes"
+	"context"
+	"encoding/gob"
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"path/filepath"
+	"time"
+
+	"github.com/src-d/metadata-retrieval/github"
+
+	"github.com/motemen/go-loghttp"
+	"golang.org/x/oauth2"
+	"gopkg.in/src-d/go-log.v1"
+)
+
+// This script will use the downloader to crawl using the graphql API a repository (default: src-d/gitbase)
+// and an organization (default: src-d) and will record requests (graphql queries) and responses (json body data)
+// into a map stored in a gob file.
+// NB: script is assumed to run from the project root (for correct file storage)
+func main() {
+	var (
+		org  string
+		repo string
+	)
+	flag.StringVar(&org, "org", "src-d", "a GitHub organization")
+	flag.StringVar(&repo, "repo", "gitbase", "a GitHub repository")
+	flag.Parse()
+
+	// Variables to hold graphql queries (keys) and responses (values)
+	var (
+		reqResp map[string]string = make(map[string]string)
+		query   string
+		ctx     context.Context
+	)
+	ctx = context.Background()
+	// Create a with authentication and add logging (recording)
+	client := oauth2.NewClient(
+		ctx,
+		oauth2.StaticTokenSource(
+			&oauth2.Token{AccessToken: os.Getenv("GITHUB_TOKEN")},
+		))
+
+	client.Transport = &loghttp.Transport{
+		Transport: client.Transport,
+		LogRequest: func(req *http.Request) {
+			query = cloneRequest(req)
```

Comment:
Currently, we don't have concurrent requests, but would this work in that case? I mean using `query` as the key of the immediately next response that will arrive.

Reply:
Oh, you're downloading a single repo, and I don't think that parallelization more fine-grained than repo will change soon, so just ignore this.

Reply:
Yes I am testing `DownloadRepository` and `DownloadOrganization` methods only (and their subsequent calls). If more parallelization arrives, we can add more tests to account for that.